engine/crates/integration-tests/src/engine_v1/gateway.rs

```diff
@@ -64,19 +82,21 @@ impl GatewayBuilder {
         };
 
         // This AuthService is used to authenticate "user" requests
-        let auth = AuthService::new_v1(
+        let auth = self.auth_service.unwrap_or(AuthService::new_v1(
```

Comment:
Not super important, but should this be `unwrap_or_else` to avoid always creating the AuthService?